packages/react-components/react-button/src/components/CompoundButton/CompoundButton.tsx

```diff
@@ -12,6 +13,8 @@ export const CompoundButton: ForwardRefComponent<CompoundButtonProps> = React.fo
   const state = useCompoundButton_unstable(props, ref);
 
   useCompoundButtonStyles_unstable(state);
+  const componentStyles = useCustomStyleHooks_unstable();
```

Comment:
this const confused me at first. I thought componentStyles was the base styles, not custom styles. Can we use customComponentStyles or something like that? Or can we just skip the const and destructure the context

const {useCompoundButtonStyles_unstable} = useCustomStyleHooks_unstable();

I've also seen context patterns that minimize re-renders when part of the context changes. Are we using that here?

Reply:
I'll update to use destructuring although I'll have to deal with a name conflict. I'm following the same pattern as the other context-related parts of FluentProvider and shared-contexts.  I think the re-render minimization was around parent/child renders with selectors.